sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/BatchCommitLog.scala

```diff
@@ -45,33 +45,39 @@ import org.apache.spark.sql.SparkSession
 class BatchCommitLog(sparkSession: SparkSession, path: String)
   extends HDFSMetadataLog[String](sparkSession, path) {
 
+  import BatchCommitLog._
+
+  def add(batchId: Long): Unit = {
+    super.add(batchId, EMPTY_JSON)
+  }
+
+  override def add(batchId: Long, metadata: String): Boolean = {
+    throw new UnsupportedOperationException(
```

Comment:
If we want metadata down the road, we will refactor this class anyways to make it not HDFSMetadataLog[String] but HDFSMetadataLog[SomeCaseClass] (like OffsetSeqLog).